src/unix/core.c

```diff
@@ -1906,9 +1933,43 @@ unsigned int uv_available_parallelism(void) {
 
   return (unsigned) rc;
 #else  /* __linux__ */
-  long rc;
+  long rc = -1;
+# if __FreeBSD__
+  cpuset_t set;
+
+  memset(&set, 0, sizeof(set));
+
+  if (0 == cpuset_getaffinity(CPU_LEVEL_WHICH, CPU_WHICH_PID, -1, sizeof(set), &set))
+    rc = CPU_COUNT(&set);
+# elif __NetBSD__
+  cpuset_t *set = cpuset_create();
+  if (set != NULL) {
+    if (0 == sched_getaffinity_np(getpid(), sizeof(set), &set))
+      rc = CPU_COUNT(&set);
+    cpuset_destroy(set);
+  }
+# elif defined(__OpenBSD__) || defined(__APPLE__)
+  int nprocs;
+  size_t len = sizeof(nprocs);
+  static int mib[][2] = {
+#  ifdef HW_NCPUONLINE
+    { CTL_HW, HW_NCPUONLINE },
+#  endif
+    { CTL_HW, HW_NCPU }
```

Comment:
According to the macOS man page:

```
     hw.ncpu
             The number of cpus. This attribute is deprecated and it is recommended that hw.logicalcpu, hw.logicalcpu_max,
             hw.physicalcpu, or hw.physicalcpu_max be used instead.
```

Not sure if that has been there forever or not.

Reply:
Hmm, I missed that, but found yet another comment in the `sysctl.h` header:

```
 *   hw.ncpu                   - The maximum number of processors that could be available this boot.
 *                               Use this value for sizing of static per processor arrays; i.e. processor load statistics.
 *
 *   hw.activecpu              - The number of processors currently available for executing threads.
 *                               Use this number to determine the number threads to create in SMP aware applications.
 *                               This number can change when power management modes are changed.
 *
 *   hw.physicalcpu            - The number of physical processors available in the current power management mode.
 *   hw.physicalcpu_max        - The maximum number of physical processors that could be available this boot
 *
 *   hw.logicalcpu             - The number of logical processors available in the current power management mode.
 *   hw.logicalcpu_max         - The maximum number of logical processors that could be available this boot
```
 
I am guessing `hw.activecpu` should be the right one to pick followed by `hw.logicalcpu`?

Reply:
Sounds good!

Reply:
Done; there's a lot of fixup commits (as I fixed the individual platforms) to make the review easier, I will squash them when you say so.